### PR TITLE
Add event Emitters for HUBBLE-244

### PR DESCRIPTION
**Pull request by dm-becker**

### CODE REVIEW

File: @hubblecommerce/hubble/core/components/checkout/PaymentMethods.vue [hunk 1]
```diff
@@ -102,6 +102,11 @@ export default {
                 context.emit('processing', false);
                 context.emit('payment-error', false);
                 context.emit('payment-changed', currentMethodObj.value);
+                $nuxt.$emit('checkout-payment-method-changed', {
+                    checkout: {
+                        actionField: { step: 2, option: currentMethodObj.value.name }
```
Comment (dm-heinze) [inline @hubblecommerce/hubble/core/components/checkout/PaymentMethods.vue:107]: Remove checkout step to keep data apstract: https://developers.google.com/analytics/devguides/collection/ua/gtm/enhanced-ecommerce#checkoutstep

File: @hubblecommerce/hubble/core/components/detail/ViewProduct.vue [hunk 2]
```diff
@@ -100,6 +100,13 @@ export default {
 
             return path;
         },
+        category: function () {
+            if (Object.keys(this.breadcrumb).length > 0) {
+                return this.breadcrumb[Object.keys(this.breadcrumb)[Object.keys(this.breadcrumb).length - 2]].name;
```
Comment (dm-heinze) [inline @hubblecommerce/hubble/core/components/detail/ViewProduct.vue:105]: Does this work with all kind of breadcrumb structure?

File: @hubblecommerce/hubble/core/pages/checkout/index.vue [hunk 3]
```diff
@@ -183,6 +183,30 @@ export default {
                 // Init payment
                 paymentResponse = await this.handlePayment({ orderId: order.data.id, dataBag: dataBag });
 
+                let productArray = [];
```
Comment (dm-heinze) [inline @hubblecommerce/hubble/core/pages/checkout/index.vue:186]: rename to plural productsArray

**Review by dm-heinze — CHANGES_REQUESTED** [review]

File: @hubblecommerce/hubble/core/components/detail/ViewProduct.vue (hunk 2 above)
Comment (dm-becker) [reply @hubblecommerce/hubble/core/components/detail/ViewProduct.vue:105]: Yes, it always returns the parent category.

**Review by dm-heinze — APPROVED** [review]

**Pull request merged**